lib/web/app/transport.go

```diff
@@ -246,7 +246,10 @@ func (t *transport) DialContext(ctx context.Context, _, _ string) (conn net.Conn
 		appServer := servers[i]
 		conn, err = dialAppServer(ctx, t.c.proxyClient, t.c.identity.RouteToApp.ClusterName, appServer)
 		if err != nil && isReverseTunnelDownError(err) {
-			t.c.log.Warnf("Failed to connect to application server %q: %v.", appServer, err)
+			t.c.log.WithFields(logrus.Fields{
+				"app_server": appServer.GetName(),
+				"app":        appServer.GetApp().GetName(),
+			}).Warnf("Failed to connect to application server: %v", err)
```

Comment:
```suggestion
			}).WithError(err).Warn("Failed to connect to application server")
```

Reply:
I really don't like `WithError` for warnings because the full stack trace catches your attention and makes it look like an error, but if you insist.

Reply:
I'm fine with both.
I tend to default to use `WithError` every time there's an error that is being logged